os/drivers/audio/audio_null.c

```diff
@@ -1000,8 +1009,24 @@ static int null_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned long
 	case AUDIOIOC_STOPPROCESS: {
 		auddbg("set stop process!!\n");
 #ifdef CONFIG_AUDIO_PROCESSING_FEATURES
-		priv->speech_state = AUDIO_NULL_SPEECH_STATE_NONE;
-		priv->process_terminate = true;
+		if ((uint8_t)arg == AUDIO_SD_KEYWORD_DETECT) {
+			priv->keyword_detect = false;
+		} else if((uint8_t)arg == AUDIO_SD_ENDPOINT_DETECT) {
+			priv->endpoint_detect = false;
+		}
+
+		if (!priv->keyword_detect && !priv->endpoint_detect) {
```

Comment:
Or we can create thread at register, destroy at unregister.
Then it may need a sem_wait.